File: 11-container-with-most-water/container-with-most-water.java
class Solution {
    public int maxArea(int[] h) {
        int i=0;
        int j=h.length-1;
        int maxi=Integer.MIN_VALUE;
        while(i<j){
            int wid=j-i;
            int hei=Math.min(h[i],h[j]);
            int area=wid*hei;
            maxi=Math.max(area,maxi);
            if(h[i]<h[j]){
                i++;
            }else{
                j--;
            }
        }
        return maxi;
    }
}

File: 11-container-with-most-water/container-with-most-water.py
class Solution(object):
    def maxArea(self, h):
        """
        :type height: List[int]
        :rtype: int
        """
        i=0
        j=len(h)-1
        maxi=0
        while(i<j):
            mini=min(h[i],h[j])
            area=mini*(j-i)
            maxi=max(maxi,area)
            if(h[i]<h[j]):
                i+=1
            else:
                j-=1
        return maxi

        